parcels/interaction/base_hash.py

```diff
@@ -0,0 +1,167 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+
+class BaseHashNeighborSearch(ABC):
```

Comment:
I can already say that I generally like here the use of docstrings for documentation. very neat indeed.